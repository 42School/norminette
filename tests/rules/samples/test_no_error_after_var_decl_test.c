void	test(void)
{
	int	x;
	{
		printf("Ok?");
	}
}

void	test2(void)
{
	int	x;
	{{ printf(1); }}
}

void	test3(void)
{
	int	y;
	(printf(1));
}

void	test4(void)
{
	int	y;
	if (1)
	{
		printf(1);
	}
}

void	test5(void)
{
	int	x, y, z;
	{
		printf("Not ok");
	}
}
