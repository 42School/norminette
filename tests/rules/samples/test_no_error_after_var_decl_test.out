[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 1":
		<VOID> <TAB> <IDENTIFIER=test> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 2":
		<LBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 3":
		<TAB> <INT> <TAB> <IDENTIFIER=x> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 4":
		<TAB> <LBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFunctionCall[0m In "ControlStructure" from "Function" line 5":
		<TAB> <TAB> <IDENTIFIER=printf> <LPARENTHESIS> <STRING="Ok?"> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "ControlStructure" from "Function" line 6":
		<TAB> <RBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 7":
		<RBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 8":
		<NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 9":
		<VOID> <TAB> <IDENTIFIER=test2> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 10":
		<LBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 11":
		<TAB> <INT> <TAB> <IDENTIFIER=x> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 12":
		<TAB> <LBRACE> 
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "ControlStructure" from "Function" line 12":
		<LBRACE> <SPACE> 
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFunctionCall[0m In "ControlStructure" from "ControlStructure" line 12":
		<IDENTIFIER=printf> <LPARENTHESIS> <CONSTANT=1> <RPARENTHESIS> <SEMI_COLON> <SPACE> 
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "ControlStructure" from "ControlStructure" line 12":
		<RBRACE> 
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "ControlStructure" from "Function" line 12":
		<RBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 13":
		<RBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 14":
		<NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 15":
		<VOID> <TAB> <IDENTIFIER=test3> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 16":
		<LBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 17":
		<TAB> <INT> <TAB> <IDENTIFIER=y> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsDeclaration[0m In "Function" from "GlobalScope" line 18":
		<TAB> <LPARENTHESIS> <IDENTIFIER=printf> <LPARENTHESIS> <CONSTANT=1> <RPARENTHESIS> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 19":
		<RBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 20":
		<NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 21":
		<VOID> <TAB> <IDENTIFIER=test4> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 22":
		<LBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 23":
		<TAB> <INT> <TAB> <IDENTIFIER=y> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsControlStatement[0m In "Function" from "GlobalScope" line 24":
		<TAB> <IF> <SPACE> <LPARENTHESIS> <CONSTANT=1> <RPARENTHESIS> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "ControlStructure" from "Function" line 25":
		<TAB> <LBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFunctionCall[0m In "ControlStructure" from "Function" line 26":
		<TAB> <TAB> <IDENTIFIER=printf> <LPARENTHESIS> <CONSTANT=1> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "ControlStructure" from "Function" line 27":
		<TAB> <RBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 28":
		<RBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 29":
		<NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 30":
		<VOID> <TAB> <IDENTIFIER=test5> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 31":
		<LBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 32":
		<TAB> <INT> <TAB> <IDENTIFIER=x> <COMMA> <SPACE> <IDENTIFIER=y> <COMMA> <SPACE> <IDENTIFIER=z> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 33":
		<TAB> <LBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsFunctionCall[0m In "ControlStructure" from "Function" line 34":
		<TAB> <TAB> <IDENTIFIER=printf> <LPARENTHESIS> <STRING="Not ok"> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "ControlStructure" from "Function" line 35":
		<TAB> <RBRACE> <NEWLINE>
[36mtest_no_error_after_var_decl_test.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 36":
		<RBRACE> <NEWLINE>
test_no_error_after_var_decl_test.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
Error: NL_AFTER_VAR_DECL    (line:   4, col:   1):	Variable declarations must be followed by a newline
Error: NL_AFTER_VAR_DECL    (line:  12, col:   1):	Variable declarations must be followed by a newline
Error: BRACE_SHOULD_EOL     (line:  12, col:   5):	Expected newline after brace
Error: TOO_FEW_TAB          (line:  12, col:   6):	Missing tabs for indent level
Error: BRACE_SHOULD_EOL     (line:  12, col:   7):	Expected newline after brace
Error: TOO_FEW_TAB          (line:  12, col:   8):	Missing tabs for indent level
Error: TOO_MANY_INSTR       (line:  12, col:   8):	Too many instructions on a single line
Error: BRACE_SHOULD_EOL     (line:  12, col:  19):	Expected newline after brace
Error: TOO_FEW_TAB          (line:  12, col:  19):	Missing tabs for indent level
Error: TOO_MANY_INSTR       (line:  12, col:  19):	Too many instructions on a single line
Error: TOO_FEW_TAB          (line:  12, col:  20):	Missing tabs for indent level
Error: TOO_MANY_INSTR       (line:  12, col:  20):	Too many instructions on a single line
Error: NL_AFTER_VAR_DECL    (line:  18, col:   1):	Variable declarations must be followed by a newline
Error: NL_AFTER_VAR_DECL    (line:  24, col:   1):	Variable declarations must be followed by a newline
Error: MULT_DECL_LINE       (line:  32, col:  10):	Multiple declarations on a single line
Error: MULT_DECL_LINE       (line:  32, col:  13):	Multiple declarations on a single line
Error: NL_AFTER_VAR_DECL    (line:  33, col:   1):	Variable declarations must be followed by a newline
